### ISSUE TITLE
No side-by-side editing button

### ISSUE THREAD

**Issue by fixer1234** [soscripted/sox#62]
**Installed Version: 1.0.3  |  Firefox/Greasemonkey (Windows 7)**

**Expected Behavior**
How do you expect the feature to work?

Assume the side-by-side editing "button" is similar to that in the standalone script (text in the editing toolbar).

**Actual Behavior**
How is the feature currently behaving?

No button/link I can find anywhere to toggle side-by-side editing

**Error Logs**
SOX errors are logged to the browser console, if possible, please post those here.

Not familiar with this.

**Steps to reproduce**
What steps did you take to reproduce the issue?

Installed script, tried to edit both answers and questions (no side-by-side option on either).  Haven't exhaustively checked the other numerous features, but many new features and the settings gears are visible, so the script successfully installed.

**Comment by shu8** [comment]
@fixer1234 could you try searching for 'side' in the search box in the settings panel and make sure `Add a button to the editor toolbar to start side-by-side editing` is checked.

Also, try pressing <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>j</kbd> to open the 'console'. Could you copy/paste anything you see there to help debugging this please? 

thanks!! :)

**Comment by fixer1234** [comment]
The first thing I verified was that the feature was checked (also verified that the computer was plugged in :-} ).  Here's the console content:

Use of Mutation Events is deprecated. Use MutationObserver instead. options-content.js:36:0
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create panel.js:5901:3
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/ panel.js:3610:3
not well-formed messages.json:1:1
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
1461536826919	Services.HealthReport.HealthReporter	WARN	Saved state file does not exist.
**ReferenceError: waitForKeyElements is not defined sox.features.js:1275:13**
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] qa.sockets.stackexchange.com
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. jquery.min.js:3:6598
Content Security Policy: Directive 'frame-src' has been deprecated. Please use directive 'child-src' instead. <unknown>
call to eval() blocked by CSP <unknown>
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
Empty string passed to getElementById(). github-f83ae2f0d483dddc8f1894b3cc2709ff1db48511ddc8d054ed639c52c80d54c9.js:6:20273
Empty string passed to getElementById(). frameworks-989d38f4923487cfd9bb7181fd4f1a7ae1c279f2a1318efe9c593f34de850870.js:9:31625
Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src https://assets-cdn.github.com"). sox.helpers.js:2:0
Empty string passed to getElementById(). frameworks-989d38f4923487cfd9bb7181fd4f1a7ae1c279f2a1318efe9c593f34de850870.js:9:31625
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create frameworks-989d38f4923487cfd9bb7181fd4f1a7ae1c279f2a1318efe9c593f34de850870.js:1:16542
Empty string passed to getElementById(). frameworks-989d38f4923487cfd9bb7181fd4f1a7ae1c279f2a1318efe9c593f34de850870.js:9:31625
Key event not available on GTK2: key="u" modifiers="accel,shift" browser.xul
Key event not available on GTK2: key="u" modifiers="accel shift" browser.xul
Key event not available on GTK2: key="u" modifiers="control shift" browser.xul
Key event not available on some keyboard layouts: key="c" modifiers="accel,alt" browser.xul
Key event not available on some keyboard layouts: key="i" modifiers="accel,alt,shift" browser.xul

I see the same waitForKeyElements reference, so it looks like the issue may be the same.  The next sentence was added to this post by the system:

Key event not available on some keyboard layouts: key="z" modifiers="accel, alt, shift" browser.xul
TypeError: keyset is null bootstrap.js:

**Comment by shu8** [comment]
@fixer1234 thanks a lot! it seems it is a problem with waitforkeyelements again :(

I think I have a fix and will ping you again tomorrow when I've got a bit of time to actually do it :P 

Sorry for the delay! and thanks again! :)

**Comment by shu8** [comment]
@fixer1234 sorry for the extreme delay!! had a busy week :/

Please could you uninstall SOX and the original side by side userscript, and then install [this version of SOX](https://gist.github.com/shu8/0a0f2dad4ceeeb514625ba003c3b255e/raw/d71295644ca3634215a541b3e3ce6025bf9f8532/sox_debug.user.js), where I've added waitforkeyelements sightly differently. 

hopefully, this will work! because I can't actually reproduce this! :/

**Comment by catb0t** [comment]
Hi @fixer1234! From the console content, it seems like you possibly have other scripts or add-ons installed that might be interfering with SOX's, ahem, *hacky ways*. :P If you can, try disabling or uninstalling those.

**Comment by shu8** [comment]
@catb0t @fixer1234 ah this is Firefox!! sorry - I didn't notice that :/

I'll check again on FF and try to push a fix later today.

sorry for the delay, again! :)

**Comment by shu8** [comment]
@catb0t @fixer1234 hmm.. I still can't reproduce this in Firefox! Could you both try disabling any other userscripts you have, leaving the version of SOX I linked to installed?

Something must be clashing with it!! :/

**Comment by fixer1234** [comment]
There are over 50 features.  Roughly half are of interest to me and many of those are "must haves" that really should be incorporated into the SE sites.  When I first loaded SOX, there were many that didn't work (too many for something that has been released to the public), so I assumed that there are a few common denominator problems, some of which might disappear when SBS was fixed, and I wouldn't be surprised if the major cause is other add-ons I have loaded.  Figured I would wait until the SBS problem was solved to see the effect before reporting anything.

Yesterday, I uninstalled the old SOX and separate SBS and installed the SOX version from the link a few comments above.  The result was that SBS works, along with a couple of other features that previously didn't.  But some features that previously worked now don't, and a few things now work differently.

The only other userscript is the auto-comments. SOX has a similar sounding feature, but that would work only if it can use my customized comments.  I did disable auto-comments as a test (including restarting FF), but that didn't affect anything in SOX.  Auto-comments had indigestion with a few add-ons (Adblock Plus and Ghostery), so those were already disabled on SE sites and aren't the culprits.

I'll do some testing to document what does and doesn't work now, and try to identify any add-ons that could be the problem.  It could take a few days.

**Comment by enki-code** [comment]
@fixer1234 I sincerely apologize for the broken features, most of them have been migrated from an older project and we've changed a lot since then without really going through all of the features to rewrite them and verify that everything is working. That's our fault.

In the the v1.0.4 milestone I am going to go through each individual feature, refactor and optimize everything. 

Some features may be depreciated in the process and some others will be added.

Also, feel free to submit pull-requests, @shu8 and I work on this when we have free time, which isn't much.

Again, sorry for the inconvenience and broken features, we'll do our best to get everything working.

**Comment by fixer1234** [comment]
Finished a preliminary look at the features and summarized the results in the attached PDF.  I haven't yet experimented with disabling add-ons.  That could take some time, so let me know if you have any insights that might reduce the job.
[SOX Test.pdf](https://github.com/soscripted/sox/files/243972/SOX.Test.pdf)

**Comment by shu8** [comment]
@fixer1234 thank you very much for making that file! I'm going to go through the features and will get back to you ASAP!

sorry again for the troubles!

**Comment by shu8** [comment]
@fixer1234 Please see this edited version, with a new column with a few comments on everything that isn't working: [SOX.Test.pdf](https://github.com/soscripted/sox/files/244224/SOX.Test.pdf)

Regarding the SBS issue, the code is exactly the same for the standalone version and the SOX version, so it should work! 

Please see if the comments on the file help you fix anything :)

**Comment by fixer1234** [comment]
Partial success.  Revised results (just the previous problem items):
[SOX Test Updates Only.pdf](https://github.com/soscripted/sox/files/244534/SOX.Test.Updates.Only.pdf)

My view of the example for Make title edits show seperately rather than merged in edit suggestions:
![su screenshot2](https://cloud.githubusercontent.com/assets/10065766/14947712/650e1ff2-1005-11e6-8536-664aca4ac16b.png)
My understanding from the features link was that interspersing the additions and deletions like this was what the feature was intended to fix.

There were a few console items that look related:

```
TypeError: Type does not implements method: modelFor#0buulgteh2g core.js:116:1
TypeError: $node.find(...).eq(...).attr(...) is undefined sox.features.js:1352:28
TypeError: Type does not implements method: modelFor#0buulgteh2g core.js:116:1
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. jquery.min.js:3:6598
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] qa.sockets.stackexchange.com
TypeError: Type does not implements method: modelFor#0buulgteh2g core.js:116:1
no element found github-cloud.s3.amazonaws.com:1:1
Content Security Policy: The page's settings blocked the loading of a resource at https://github-cloud.s3.amazonaws.com/ ("base-uri https://github.com"). <unknown>
no element found github-cloud.s3.amazonaws.com:1:1
TypeError: keyset is null bootstrap.js:34:5
```

There were some vwof plugin errors (like: vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined), which I assume are unrelated.

**Comment by shu8** [comment]
@fixer1234 Thanks, please see this updated version: 
[SOX.Test.Updates.Only.pdf](https://github.com/soscripted/sox/files/245353/SOX.Test.Updates.Only.pdf)

Please also try installing this edited version that should fix the enhanced editor issue: https://gist.github.com/shu8/8bf4a5dfbc0f12fdef42717fa80b0259/raw/22d709a2c491c2086f846c8492a859a1401e760b/fixer%2520debug 

---

Now, provided everything else works, it seems your issues are all to do with the editor toolbar (list+kbd buttons, SBS, enhanced editor). I can only think that something is messing with the toolbar that makes the script unable to add the right buttons.

Could you try checking the console again please? I know I keep asking you to do that, but it is literally the only thing that can actually tell us what is wrong so we can fix it! Sorry for the time this is taking; I hope it can all be fixed soon! :)

**Comment by fixer1234** [comment]
@shu8, I clicked on the github link but it just displays a listing of the script.

BTW, on the make title edits show separately issue, the feature is suddenly working on its own (no change on my part).  It isn't clear whether the script gets code at runtime and that's been fixed, or this is another example of intermittent performance (how can you even debug that?).

**Comment by shu8** [comment]
@fixer1234 sorry, forgot to change the extension. [This link should work](https://gist.github.com/shu8/8bf4a5dfbc0f12fdef42717fa80b0259/raw/ff8c3d25b33ff05b2d8b3f53393d68dda6878184/fixer%2520debug.user.js).

Also, the script does have the code remotely, which sometimes downloads to update -- it's possible you had an old version locally downloaded. Installing this new version^ should delete any previous versions and download the latest versions of everything.

**Comment by fixer1234** [comment]
@shu8 Installed the new link.  Cleared the console and updated settings, tried to get an access token.  This generated some console messages, all related only to SOX:

```
SOX settings saved: ["Appearance-metaNewQuestionAlert","Appearance-linkedToFrom","Appearance-addEllipsis","Appearance-standOutDupeCloseMigrated","Appearance-scrollToTop","Appearance-addAuthorNameToInboxNotifications","Appearance-markEmployees","Appearance-colorAnswerer","Appearance-tabularReviewerStats","Appearance-fixedTopbar","Appearance-hideCommunityBulletin","Appearance-hideHotNetworkQuestions","Appearance-renameChat","Appearance-metaChatBlogStackExchangeButton","Appearance-answerTagsSearch","Comments-confirmNavigateAway","Comments-commentReplies","Comments-showCommentScores","Comments-commentShortcuts","Comments-autoShowCommentImages","Editing-addSBSBtn","Editing-editReasonTooltip","Editing-kbdAndBullets","Editing-enhancedEditor","Editing-titleEditDiff","Flags-flagPercentages","Flags-flagOutcomeTime","Unsorted-downvotedPostsEditAlert","Unsorted-alwaysShowImageUploadLinkBox","Unsorted-shareLinksMarkdown","Unsorted-linkedPostsInline","Unsorted-parseCrossSiteLinks"] fixer%252520debug.user.js:92:13
The connection to ws://qa.sockets.stackexchange.com/ was interrupted while the page was loading. full.en.js:1:5130
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. jquery.min.js:3:6598
no element found 381b:1:1
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] qa.sockets.stackexchange.com
ReferenceError: waitForKeyElements is not defined sox.features.js:1313:21
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create panel.js:5813:3
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/ panel.js:3522:3
not well-formed messages.json:1:1
```

I haven't started any kind of exhaustive checking, yet, but re: edit issues: the only changes to the default edit window is Find&Replace+AutoCorrect.  No SBS, KBD, etc.

Question on access tokens: it looks like inbox notification on question edit is the only (selected) feature requiring a token.  That still says I need to be on SO to get it.  Is that the only site where the feature will work?  I'm not a programmer, so I don't use SO.  However, I'm happy to do some diagnostics for you if it will be useful.  If so, what is the expected behavior and how does the button work?  (is it a toggle or each click will attempt to get a token?  What feedback am I looking for to know if anything resulted from the click?)

**Comment by shu8** [comment]
@fixer1234 thanks for all the console dumps! They really do help.

It seems the issue is the waitforkeyelements again. weird, but I've changed the version again to include it in a different way. Please (sorry for doing this so many times!) install [this SOX version](https://gist.github.com/shu8/8bf4a5dfbc0f12fdef42717fa80b0259/raw/e9e2f8331913e25a55dc5296862e42670e3d2971/fixer%2520debug.user.js) now. It includes waitforkeyelements in a different way which should work!

---

About your access token question: you only need to be on SO to actually get the access token. Once the access token is saved, it will work on any site. So, what should happen is:

- you go on SO.com
- you click 'new access token' / 'get access token'
- a window pops up -- you click approve (and login if neccessary)
- the window closes, and your new access token appears in the SOX settings dialog, where the link to get one was

then it should work: clicking on 'notify on edit' should make it turn green (will make that more obvious soon), and an edit should trigger a notification. [Here's an answer](http://meta.stackexchange.com/a/276840/260841) that describes how it works (note; the eye icon has been replaced with the 'notify on edit' text!).

---

I have reproduced your bug with the StackExchange button -- it's actually an issue with the hovering over the button thing. This will be fixed in 1.0.4.

---

Finally, I still think it could be an add on you've installed that's not making the SBS and other features work! I think that more are working on your other computer because that 'thing' isn't there. Please do try to create a new FF profile, and *just* install Greasemonkey and SOX (with the link I posted at the beginning of this comment). Then, hopefully, everything should work. I say this because I can't reproduce the reference errors to waitforkeyelements, so there must be something blocking it.

**Comment by fixer1234** [comment]
@shu8: Loaded the revised script and just took a quick look at the edit box and tokens.  The tokens link didn't work.  The edit box has Find/Replace and Auto Correct, and the SBS link (only additional features).  SBS doesn't work.  It provides a centered edit box followed by the preview.  Console contents:

```
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create frameworks-90fadde68d68d108f0318274e0f72c8e36717b821c919eb6884c738f6bef8470.js:1:16542
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
The connection to wss://live.github.com/_sockets/MTAwNjU3NjY6NzQzOGYwMTBlY2Q4ZjdhOGExZGJjOTNiMWU4MTdhYmU6ZWQ2NTk3NDBmNzAzNDk4NzllMjc3MDZmMzc3OTEwMjAzZTc2Zjg5NGI5ZjcyZjQxNjlmOGE5ZWQxOTEwYTk0OA==--ea5337ed952962ce10895e9b4ee02d633c59c05a was interrupted while the page was loading. github-b34f54eabccd0ffa90e322cffdb0db239d492b9a986092f1aa602c5cf84d36b4.js:4:29239
TypeError: keyset is null bootstrap.js:34:5
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create panel.js:5813:3
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create panel.js:5813:3
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/ panel.js:3522:3
not well-formed messages.json:1:1
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] qa.sockets.stackexchange.com
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. jquery.min.js:3:6598
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
g is null jquery-2.1.4.min.js:3:19226
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
TypeError: l is not a function all.js:1:491
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
TypeError: l is not a function all.js:1:491
TypeError: $node.find(...).eq(...).attr(...) is undefined sox.features.js:1354:28
vwof plugin, exception in ScreenWaveMedia: TypeError: video_data is undefined vwof.js:67:0
Key event not available on GTK2: key="u" modifiers="accel,shift" browser.xul
Key event not available on GTK2: key="u" modifiers="accel shift" browser.xul
Key event not available on GTK2: key="u" modifiers="control shift" browser.xul
Key event not available on some keyboard layouts: key="c" modifiers="accel,alt" browser.xul
Key event not available on some keyboard layouts: key="i" modifiers="accel,alt,shift" browser.xul
Key event not available on some keyboard layouts: key="z" modifiers="accel, alt, shift" browser.xul
TypeError: keyset is null bootstrap.js:34:5
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create panel.js:5813:3
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/ panel.js:3522:3
not well-formed messages.json:1:1
```
I'll play with a new profile in the next day or so.

**Comment by fixer1234** [comment]
@shu8: BTW, this is probably what you were referring to regarding the issue with some image links in comments showing as a box with an icon.  The examples were cases where the link isn't to an actual image, with an image file extension in the link, but to an imgur page that resembles a slideshow, and the image is displayed in that.  Example: 2nd pablosan comment on [How do I create an Outlook rule for moving all messages flagged on a particular day?](http://superuser.com/a/923282/364367) contains imgur.com/kRhgdQq.

**Comment by shu8** [comment]
@fixer1234 thanks for the example -- that issue will be fixed in v1.0.4.

Also, could you please screenshot what you see when you click the SBS icon? From what I understand, it is doing *something*, just not what it should...

I think I may be on to something, so this might fix it! Thanks for your patience! :)

**Comment by fixer1234** [comment]
@shu8 A couple of interesting developments that might shed some light on something.

First, as noted in my previous comment, the kbd feature wasn't showing.  I haven't changed anything, but now it's there.  That seems to point to some external code and either it, or access to it, has changed independent of what is in the SOX script.

Interesting discover on SBS.  Here is an image of the feature activated:

![capture1](https://cloud.githubusercontent.com/assets/10065766/15094702/0f2f61fc-147b-11e6-8dc8-247b0d1cf301.PNG)

Notice that the edit window is more or less centered in the window and the preview window is below it, staggered a little to the right.  However, when I actually use the Find and Replace feature, SBS displays correctly:

![capture2](https://cloud.githubusercontent.com/assets/10065766/15094715/6d498394-147b-11e6-9f33-ef63701f946b.PNG)

If I click on Find and Replace again to toggle it off, SBS returns to the view in the first image.

On my other computer, which is still running an earlier version of 1.0.4 DEV, the SBS button is there, but Find and Replace/AutoCorrect and KBD don't appear.  On that one, SBS works correctly.  So it looks like some kind of conflict between SBS and the Find&Replace modules.

**Comment by shu8** [comment]
@fixer1234 aha! Thanks a lot!! I think I've got it!! :D 

I'm going to try fix this, but I've got a pretty busy week ahead of me, so it might not be until next weekend when I get back to you :/

sorry!

**Comment by shu8** [comment]
@fixer1234 I've made a few changes that should fix all your issues now! :)

Please install [this version](https://github.com/soscripted/sox/raw/dev/sox.user.js).

The access token issue, however, remains. I can reproduce it on Firefox but am unable to solve it right now :(

That said, @enki-code and I have been refactoring the script heavily, and we will be changing how the access tokens work in the next version. Till then, I hope your other issues have been fixed, and thanks for staying with us all this time! 

Please confirm whether these issues are fixed for you:

- SBS
- imgur images in comments
- add link from web input
- KBD and list buttons

Thanks

**Comment by fixer1234** [comment]
@shu8: Quick test results:

SBS: works properly with Find/Replace **not** selected.  However, when Find/Replace is selected, the edit window remains in the proper position and the preview window is on the right, but the preview window drops below the edit window:

![sbs with find replace](https://cloud.githubusercontent.com/assets/10065766/15226322/36e4f486-1850-11e6-8366-868222179aca.png)

KBD/List: notice in the image that these buttons still don't appear.

Images in comments: working.  Image files display, non-images show the link.

link from web input: Looks like you have a new UI.  Both browse and web allow selection or entry of file/URL, but neither insert button works.

So progress and partial success, but no cigar, yet.

**Comment by shu8** [comment]
@fixer1234 sorry about that!

Please install this [updated version](https://github.com/soscripted/sox/raw/dev/sox.user.js)

- fixed the SBS, again
- added a delay for the kbd and list buttons, so should appear now
- yay for images!
- the new UI is part of the enhanced editor feature (which I've also fixed, so they should work now)

**Comment by shu8** [comment]
@fixer1234 regarding the edit comment. could you try pressing Ctrl + Shift + Q and then clicking the Network panel, and then trying to hover over the date?

I have a feeling you're being throttled by the API; when it doesn't work, a network request should appear in the Network tab, and it should be red on the left if it didn't work. Can you screenshot it if it is red?

thanks

**Comment by fixer1234** [comment]
@shu8: Retest: 

SBS: Working!

KBD/List: Looks like it's working (success on several random questions)!

Image UI: Working but needs a minor fix.  There needs to be at least a space between body text and the image link so that the link isn't interpreted as part of a word.  As a test, I just typed along and then inserted an image.  It didn't appear in the preview until I added separation.  Suggest starting the inserted image link string with a space to ensure that it works.

Edit comment on hover:  Let me preface this with the fact that I'm colorblind, so I'm just guessing at "red".  I picked a random question with an edit, and the hover produced only a date/time in the pop-up.  This happened to be one with a system comment ("added nn characters to body"), rather than a user comment, so at least my observation seems to be holding.  

I did the ctrl-shift-Q and then reloaded the page, which produced some output, including this:

![api](https://cloud.githubusercontent.com/assets/10065766/15229144/ad794afe-185d-11e6-80ea-55fb68e9ef8e.png)

However, hovering on the edit link reproduced the date/time-only pop-up, but no new entries in that console.

So you've got a couple of items to work on for the next release (and I'll be happy to run more tests if it will help).  In the meantime, you've earned the cigar.

**Comment by shu8** [comment]
@fixer1234 thanks; will fix the image thing.

I'll look into the comment thing a bit more and get back to you

thanks

**Comment by shu8** [comment]
@fixer1234 if you come across any more posts where it doesn't work, please could you post the link here? It would help determine what the cause is :)

**Comment by fixer1234** [comment]
@shu8 a little more anecdotal feedback:

Edit comments on hover: Based on dozens of checks, whether the pop-up displays the edit comment or just the date/time, or the predominance of which one it displays on repeat hovers, does not depend on: whether the edit is to a question or answer, the recency of the edit (seconds vs. days), or whether it is a system comment vs. user comment.  Previous observations were just coincidental.

Another random test on an edit that repeatedly popped-up just date/time: I did ctrl-shift-Q and reloaded the page.  The console detail showed the element loading activity but there was no red.  I retested hovering on the edit link many times.  A couple of times it displayed the comment (it was a user comment).  None of the hovering triggered additional activity in the console.  So if you still think this is throttling, this procedure doesn't capture it.

Title edits: Intermittent performance.  Came across a random title edit in the review queue that initially displayed deletions and additions mixed.  I reloaded the page several times and then it started displaying correctly.

Fix topbar position: I forget whether this was one slated for fixing later or you thought it was already fixed.  If I select historical comments from my profile page, the thread gets displayed with the selected comment at the top of the page.  Several lines of the comment end up hidden behind the topbar rather than being displayed below the topbar.  Clicking on question or answer posts typically don't end up partially hidden (not clear whether this is because the issue is fixed for those or there is just blank space at the top of those posts so the content ends up below the topbar).

Edit window enhancements (SBS, KBD/List, Find&Replace|AutoCorrect): these don't appear when editing a post in a review queue (tested in Suggested Edit and LQ Post review queues).

One other item I'll alert you to, although it may have nothing to do with SOX: on the main page, the number of new, unread posts is normally displayed at the top of the page and in the tab text.  It has suddenly intermittently stopped displaying this.  Around the same time I loaded the most recent SOX, the system upgraded Firefox from V45 to V46, which turned out to conflict with Oxygen themes on the system.  I had to uninstall it and manually install V45 ESR from a Mozilla tarball.  So it's back to the Firefox version that was previously working (except for the ESR, which does have a few minor differences), and the same suite of add-ons.  The issue could well relate to Firefox rather than SOX, and I'm doing some testing, but I thought I'd mention it in case there's anything in SOX that might potentially cause that.

**Comment by shu8** [comment]
@fixer1234 

- edit comments: still very weird :/ if you see it happen again, please do make a note of the URL!
- title edits: sounds like a timing issue that will be fixed in the next release
- topbar: on my todo list :)
- review queue buttons: I'll try adding it to review queues

I don't think SOX should have an effect on the tab text... there's nothing in the code that touches it! I hope it was a coincidence, because that would be very strange if SOX was causing it! :/

**Comment by fixer1234** [comment]
@shu8 

- I've confirmed that the problem with intermittent loss of the new post count (main page & tab), is not related to SOX.  This is something strange that started when FF V45 was upgraded to either V46 or V45ESR (happens in both, in 3 OSs); something changed that the SU site doesn't like.

- I did a test of the edit comments.  Procedure: I went down the main page looking for new post listings identified as modifications.  I opened the post and hovered repeatedly over the edit link.  If the edit comment showed up three successive times, I discarded the post.  Otherwise, I attempted hovering five times and recorded the result.  So all of these were ones that failed to show the comment at least some number of times (in which case it displayed the edit date and time in the pop-up).

All of this group failed 5 out of 5 times:

```
[Android virtual router](http://superuser.com/q/856197/364367) date/time 
[How do I open / extract files from a .b1 archive?](http://superuser.com/a/708725/364367) 
[AHK won't start Vagrant/Cmder on boot/reload?](http://superuser.com/q/1076955/364367) 
[Is it worth buying a SSD disk for a motherboard that only supports SATA I?](http://superuser.com/q/570409/364367) 
[Windows 10 Keyboard Repeat Rate/Delay Keeps Changing](http://superuser.com/a/1077242/364367) 
[Change Windows 10 Start Menu Icon](http://superuser.com/q/1077245/364367) 
[Catch cmd window | Can I let it write to file by std?](http://superuser.com/q/1077229/364367) 
```

This next group had some number of successes.  The coding after the link is the replications of either comment or date/time, so the first example displayed date/time on the first try, then the comment on three successive tries, then the date/time on the last try.

```
[Time machine backup on S3](http://superuser.com/a/1061900/364367) 1D, 3C, 1D
[KVM VM not visible/pingable from LAN](http://superuser.com/q/1077227/364367) 4D, 1C
[Sum numbers in filenames in a directory using command line](http://superuser.com/a/1077233/364367) 2D, 1C, 1D, 1C
[Setting up a SSL server](http://superuser.com/q/1076741/364367) 1D, 2C, 1D, 1C
[CMD window will not close if folder is closed during execution of bat file](http://superuser.com/a/1077189/364367) 1D, 4C
[Struggling with my Sony bluetooth headset dr-btn200 Windows 8.1](http://superuser.com/a/1077149/364367) 1C, 4D
```

I couldn't find any obvious characteristic differentiating the problem posts.  The script always retrieves something, but not always the comment.  Could it be a timing thing, where both the date/time and the comment are part of the value being populated or retrieved, and the speed of that (driven by other things), determines which piece gets displayed?

**Comment by fixer1234** [comment]
Documenting a strange experience in case it sheds light on anything.  I did an edit on this answer: [View menu bar &amp; bookmarks toolbar in Firefox 4 full-screen mode?](http://superuser.com/a/906137/364367).  Here is a screenshot of the edit screen:

![su edit screenshot](https://cloud.githubusercontent.com/assets/10065766/15295357/227850d6-1b5e-11e6-954a-36313ed95ea5.png)

Normally, the edit screen is at the top of the page.  Here, the edit window is positioned in front of a collection of thread elements: previous edit comment, question, and the answer being edited (which are all from different places).

Also, the SOX edit enhancements didn't all appear.  Initially, either KBD/List or SBS was missing (I forget which).  I reloaded the page and both went missing.

This is the first time either form of corruption appeared.

**Comment by shu8** [comment]
@fixer1234 please install [this modified version of SOX](https://gist.github.com/shu8/8bf4a5dfbc0f12fdef42717fa80b0259/raw/b8c374fc8fbcb1e3c271a57e5bcf78cba1e0407c/fixer%2520debug.user.js) (making sure it replaces the current version!) and go on any question page.

It should now log quite a bit to the console -- please could you copy and paste the entire console log?

---

Also, sorry but what's wrong in that screenshot? I don't understand...

And I will add the edit stuff to edits in the next version :)

**Comment by fixer1234** [comment]
@shu8 Will do.  Just open the page and copy the log, or attempt some of the problem operations and then copy the log?

Explanation of the screenshot:

- The edit operation normally replaces the original post.  Where the post used to be on the page, aligned with its voting controls, it gets replaced with the edit window, then the preview window.  The screenshot is of an edit of the last of three answers, so the post was near the bottom of the page.  In the screenshot, the edit activity is occurring at the top of the page.  First is the edit summary comment from the previous edit (not normally shown at all).  Then the top of the question, and the answer edit window is superimposed there.  

- Within the edit window, SBS and KBD/List are both missing.  The screenshot is after reloading the page.  Originally, only one of those two was missing (I forget which).  I reloaded the page to see if it would load correctly.

This was an odd-ball occurrence.  Nothing similar happened before or since, so it may be impossible to replicate.  However, I thought the constellation of symptoms might provide a clue to what might be going on behind the scenes contributing to other odd symptoms.

**Comment by shu8** [comment]
@fixer1234 sorry should have clarified.. yes ctrl+shift+j and could you copy it when you come across a post where the edit comment doesn't show up please.

and about the edit thing: it seems like you held down CTRL when clicking `edit` -- doing so opens it in a new tab and shows you the UI that you would have without having the edit privilege at 2k.

**Comment by fixer1234** [comment]
@shu8 Just as a test, I found a random post where the edit comment didn't appear.  I cleared the console, reloaded the page, then hovered on the edit history link several times.  The first two times displayed the date/time.  Third time displayed the comment.  None of the hovering added any content to the console.  Here's a screenshot:

![su screenshot temp](https://cloud.githubusercontent.com/assets/10065766/15339750/e45fca76-1c53-11e6-824d-1a04d7e4a6a3.png)

Let me know if this is capturing what you're looking for and I'll replicate it with other posts.

Re: weird edit window -- There's no way to know for sure that I didn't press Ctrl, but that does reproduce the layout.

**Comment by fixer1234** [comment]
@shu8 Another quick update.  The "Show comment score" previously worked but appears to not work any more (I'm guessing since this test release).

Let me know if the edit comment console screenshots are capturing what you were looking for, and whether to do more.

**Comment by shu8** [comment]
@fixer1234 ok, so it seems the edit comment issue is a timing issue (what a surprise :P), because the request to get the comment is going through, but it's not being applied to the tooltip every time.

this has been fixed in the refactored version, and I'll ping you in a few days (hopefully!) when that version is completely finished. We're going to try our best to make sure **everything** is working in Chrome and on Firefox (where most of our issues unfortunately come from :disappointed:).

The Show Comment Score has also been modified in the refactored version, so hopefully that will work as well.

Thanks for your help and patience. I hope the refactored version does work for you :)

**Comment by shu8** [comment]
@fixer1234 We have finally released [v2.0.0](https://github.com/soscripted/sox/raw/master/sox.user.js) just now. I'm very sorry for the long wait, but we have both had a very busy past month or two and haven't had much time to do anything on SOX.

Please could you install this new version. I hope it works well for you! :)

Thanks again for sticking with us through all this! :smile:
